frontend/model/contracts/chatroom.js

```diff
@@ -242,8 +316,16 @@ sbp('chelonia/defineContract', {
         const newMessage = createMessage({ meta, hash, data: notificationData, state })
         state.messages.push(newMessage)
       },
-      sideEffect ({ contractID, hash }) {
+      sideEffect ({ contractID, hash, meta }) {
         emitMessageEvent({ contractID, hash })
+
+        if (sbp('okTurtles.data/get', 'READY_TO_JOIN_CHATROOM')) {
+          sbp('state/vuex/commit', 'setChatRoomUnreadSince', {
+            chatRoomId: contractID,
+            messageId: hash,
+            createdDate: new Date(meta.createdDate).getTime()
+          })
+        }
```

Comment:
I have seen these 7 lines repeated verbatim in several places now. That's a DRY violation => convert to function and minimize repetition as much as possible.

Reply:
Nice catch. That's correct. Not sure why I didn't think this way.